saleor/warehouse/management.py

```diff
@@ -198,12 +199,17 @@ def increase_stock(
     function increase `quantity_allocated`. If allocation does not exist function
     create a new allocation for this order line in this stock.
     """
+    product_variant_stock_trigger = ProductVariantStockWebhookTrigger()
     stock = (
         Stock.objects.select_for_update(of=("self",))
         .filter(warehouse=warehouse, product_variant=order_line.variant)
         .first()
     )
     if stock:
+        product_variant_stock_trigger.append_stock_data(
+            stock, False, stock.quantity, quantity
+        )
+        product_variant_stock_trigger.trigger_product_variant_back_in_stock_webhook()
```

Comment:
- you're calling it before we increase stock
-  stock logic is called in a transaction which means, that there is a possibility that your hook will be called before we will update all data in DB.

Reply:
We should trigger webhook in `on_commit`